src/tests/multihost/alltests/test_sudo.py

```diff
@@ -184,9 +184,67 @@ def test_randomize_sudo_timeout(self, multihost,
                     same_intvl += 1
                 index += 1
             assert rand_intvl > same_intvl
+
+    @pytest.mark.tier2
+    def test_improve_refresh_timers_sudo_timeout(self, multihost,
+                                                 backupsssdconf,
+                                                 sssd_sudo_conf,
+                                                 sudo_rule):
+        """
+        :title: sudo: improve sudo full and smart refresh timeouts
+        :id: 3860d1b9-28fc-4d44-9537-caf28ab033c8
+        :bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1925505
+        :customerscenario: True
+        :steps:
+          1. Edit sssdconfig and specify sssd smart, full timeout option
+          2. Restart sssd with cleared logs and cache
+          3. Wait for 40 seconds
+          4. Parse logs and confirm sudo full refresh and smart refresh
+             timeout are not running at same time
+          5. If sudo full refresh and smart refresh timer are scheduled at
+             same time then smart refresh is rescheduled to the next cycle
+        :expectedresults:
+          1. Should succeed
+          2. Should succeed
+          3. Should succeed
+          4. Should succeed
+          5. Should succeed
+        """
+        tools = sssdTools(multihost.client[0])
         multihost.client[0].service_sssd('stop')
-        params = {'ldap_sudo_full_refresh_interval': '25',
-                  'ldap_sudo_smart_refresh_interval': '15',
-                  'ldap_sudo_random_offset': '5'}
-        tools.sssd_conf(domain_section, params, action='delete')
+        tools.remove_sss_cache('/var/lib/sss/db')
+        tools.remove_sss_cache('/var/log/sssd')
+        params = {'ldap_sudo_full_refresh_interval': '10',
+                  'ldap_sudo_random_offset': '0',
+                  'ldap_sudo_smart_refresh_interval': '5'}
+        domain_section = f'domain/{ds_instance_name}'
+        tools.sssd_conf(domain_section, params, action='update')
         multihost.client[0].service_sssd('start')
+        time.sleep(40)
+        logfile = f'/var/log/sssd/sssd_{ds_instance_name}.log'
+        tmout_ptrn = r"(SUDO.*Refresh.*executing)"
+        rschdl_ptrn = r"(SUDO.*Refresh.*rescheduling)"
+        regex_tmout = re.compile("%s" % tmout_ptrn)
```

Comment:
I think you should be able to use f-strings in the re.compile here too.  Or simply combine them into fewer lines if they don't exceed the line length limitation:

         regex_tmout = re.compile(r"(SUDO.*Refresh.*executing)")
         rgx_rs_tstmp = re.compile(r"(SUDO.*Refresh.*rescheduling)")

I'm ok with keeping the patterns as separate vars as you have now if you think they may need to change in the future and become too long.

Reply:
Your point is correct on using f-strings in re-compile. Here, patterns in vars is more of better coding practice.

Reply:
Shouldn't line 227 and 228 change to
`regex_tmout = re.compile(f'{tmout_ptrn}')`
`rgx_rs_tstmp = re.compile(f'{rschdl_ptrn}')`